chore: improve contributing page with docs guidelines

auroraVasconcelos opened the pull request:
<!-- ## Title: Please consider adding the [skip chromatic] flag to the PR title in case you dont need chromatic testing your changes. -->
## Description:
Closes https://github.com/solid-design-system/solid/issues/1453

## Definition of Reviewable:
<!-- *PR notes: Irrelevant elements should be removed.* -->
- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
<!-- *If this PR includes a bug fix, an E2E test is necessary to verify the change. If the fix is purely visual, ensuring it is captured within our chromatic screenshot tests is sufficient.* -->
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

auroraVasconcelos commented on CONTRIBUTING.md:
```diff
@@ -75,6 +75,41 @@
+  - All components have an introduction text descriving the functionality starting with "Used to ...", feel free to use descriptions from [shoelace.style](shoelace.style).
+  - After the introduction, the `Related components` are listed.
+  - After the `Related components` list, `Related templates` are listed.
+    - Template links in components must point to the mentioned section within the template. If the template have the name of the component link the full template.
```
What about:

>   - Links to templates within components must direct to the specified section within the template (e. g. for sd-tooltip, link to the Select with Tooltip section). If the template shares the same name as the component, link to the template page (e. g. for sd-select, link to the select template).

smfonseca replied:
Aligned on teams call.

MartaPintoTeixeira commented on CONTRIBUTING.md:
```diff
@@ -75,6 +75,40 @@
   - Provide stories for props, slots etc. in Storybook
   - Props, events, slots, parts and CSS Custom Properties should be documented in the component's source code using [JSDoc](https://jsdoc.app/). Make sure to include a description, type, and default value for each.
   - Use `.slot` and the additional utility classes classes defined in `packages/components/.storybook/preview-head.html` to mock the slot element used in Figma. These classes are integrated into the Storybook header and should exclusively be used in Storybook. Check out the 'Slot' stories for components like 'sd-dropdown' to see how these classes are used.
+- **Guidelines**:
```
Guidelines:

Navigation:
- Template file names use a "shortened name" format (e.g., use "Badge" instead of "Button with Badge").

Screenshot Tests:
- All screenshot tests should be included in {component}.test.stories.ts. Create a combination screenshot story that includes all tests.

Component Documentation:
1 ) Description: Each component must have an introductory text that describes its functionality, starting with "Used to...". You can adapt descriptions from shoelace.style.
2 ) Related components: Following the introduction, include a list of Related components (e.g., "sd-badge").
3 ) Related templates: After listing the Related components, provide a list of Related templates.
- Links to templates within components should direct to the specified section within the template (e.g., for sd-tooltip, link to the Select with Tooltip section). If the template shares the same name as the component, link directly to the template page (e.g., for sd-select, link to the select template).

MartaPintoTeixeira replied:
@auroraVasconcelos I think the post need more hierarchy to be easier to read. I left here a suggestion of splitting it into sections ex:
Navigation
Screeshot Tests
Component Docs
...

Also using numbers intead of only "-" will help understand which are the "parents or sons".

Also I would Briefly resume the topic before the quote ex:
1 ) Description: Each component must have an introductory text that describes its functionality, starting with "Used to...". You can adapt descriptions from shoelace.style.

Does it make sense to you?
Do you think you can follow this logic for the rest?

auroraVasconcelos replied:
Updated

smfonseca commented on CONTRIBUTING.md:
```diff
@@ -75,6 +75,41 @@
+  - After the introduction, the `Related components` are listed.
+  - After the `Related components` list, `Related templates` are listed.
+    - Template links in components must point to the mentioned section within the template. If the template have the name of the component link the full template.
+    - The naming of related templates must be Component Only or if pointing to a specific section Component - Component (eg: in sd-select link it to the template select, in sd-tooltip link to the section Select with Tooltip).
```
Aligned on teams call.

MartaPintoTeixeira commented on CONTRIBUTING.md:
```diff
@@ -75,6 +75,41 @@
+    - If a template for the component exists, provide a link to the template page.
+    - If the component is part of another component's template, link directly to the relevant section.
+  - For every component's "visual" feature in {component}.stories.ts (Samples):
+    - **Description:**
```
Sample Description instead of Sample

MartaPintoTeixeira replied:
Also all bold titles should be aligned to the left as much as possible.
I see Description ans the bold titles bellow a bit more to the right.

auroraVasconcelos replied:
Updated

paulovareiro29 commented on CONTRIBUTING.md:
```diff
@@ -75,6 +75,41 @@
+  - For every component's "visual" feature in {component}.stories.ts (Samples):
+    - **Description:**
+      - For each sample exists a single story with plain HTML/CSS/JS, that shows the feature.
+      - The sample has a legend with one sentence describing the functionality and what is intended to. (e. g. Use the "open" attribute to ...). Feel free to use [shoelace.style](shoelace.style) as inspiration.
```
```suggestion
      - The sample has a legend with one sentence describing the functionality and what is intended to. (e. g. Use the "open" attribute to ...). Feel free to use [shoelace](https://shoelace.style) as inspiration.
```

paulovareiro29 commented on CONTRIBUTING.md:
```diff
@@ -75,6 +75,41 @@
   - Use `.slot` and the additional utility classes classes defined in `packages/components/.storybook/preview-head.html` to mock the slot element used in Figma. These classes are integrated into the Storybook header and should exclusively be used in Storybook. Check out the 'Slot' stories for components like 'sd-dropdown' to see how these classes are used.
+- **Guidelines**:
+  - **Screenshot tests:** All screenshot tests are added in `{component}.test.stories.ts`. Create a combination screenshot story with all tests.
+  - **Description:** All components have an introduction text describing the functionality starting with "Used to ...", feel free to use descriptions from [shoelace.style](shoelace.style) as inspiration.
```
```suggestion
  - **Description:** All components have an introduction text describing the functionality starting with "Used to ...", feel free to use descriptions from [shoelace](https://shoelace.style) as inspiration.
```

MartaPintoTeixeira commented:
Can you please remove the bold for this:

Sample description:
Sample content:
Samples order:

MartaPintoTeixeira commented:
If a template for the component exists, provide a link to the template page.
If the component is part of another component's template, link directly to the relevant section.

change to

If a template for the component exists, provide a link to the template page (eg. Badge).
If the component is part of another component's template, link directly to the relevant section and name it mentiontion both of the components.(eg. Tab with Badge)

MartaPintoTeixeira commented:
Ensure you add a period at the end of all sentences.
change to:
Ensure you add a period at the end of all sentences (with exception of bullets)

auroraVasconcelos commented:
> Can you please remove the bold for this:
> 
> Sample description: Sample content: Samples order:

Updated

> If a template for the component exists, provide a link to the template page. If the component is part of another component's template, link directly to the relevant section.
> 
> change to
> 
> If a template for the component exists, provide a link to the template page (eg. Badge). If the component is part of another component's template, link directly to the relevant section and name it mentiontion both of the components.(eg. Tab with Badge)

Updated

> Ensure you add a period at the end of all sentences. change to: Ensure you add a period at the end of all sentences (with exception of bullets)

Updated
